app/lib/domain/controllers/tts_controller.dart

```diff
@@ -0,0 +1,20 @@
+import 'dart:io';
+import 'package:flutter/foundation.dart';
+import 'package:flutter_tts/flutter_tts.dart';
+import 'package:logger/logger.dart';
+
+part 'package:word_link/infrastructure/tts_repository.dart';
+
+abstract class TtsController {
+  static TtsController? _instance;
+
+  static TtsController get instance => _instance ??= _TtsRepository();
+
+  late FlutterTts? _flutterTts;
```

Comment:
If you are using late you don't need the support for null, because you are notifying the code that it will get a value but later.

Change to   `late FlutterTts _flutterTts;`
Also why is it in the controller and not the repository?, move it.